packages/@react-aria/calendar/src/useCalendarBase.ts

```diff
@@ -70,20 +70,22 @@ export function useCalendarBase(props: CalendarPropsBase & DOMProps & AriaLabeli
     selectedDateDescription
   });
 
-  // If the next or previous buttons become disabled while they are focused, move focus to the calendar body.
   let nextFocused = useRef(false);
   let nextDisabled = props.isDisabled || state.isNextVisibleRangeInvalid();
-  if (nextDisabled && nextFocused.current) {
-    nextFocused.current = false;
-    state.setFocused(true);
-  }
-
   let previousFocused = useRef(false);
   let previousDisabled = props.isDisabled || state.isPreviousVisibleRangeInvalid();
-  if (previousDisabled && previousFocused.current) {
-    previousFocused.current = false;
-    state.setFocused(true);
-  }
+  // If the next or previous buttons become disabled while they are focused, move focus to the calendar body.
+  useLayoutEffect(() => {
+    if (nextDisabled && nextFocused.current) {
+      nextFocused.current = false;
+      state.setFocused(true);
+    }
+
+    if (previousDisabled && previousFocused.current) {
+      previousFocused.current = false;
+      state.setFocused(true);
+    }
+  }, [nextDisabled, previousDisabled, state]);
```

Comment:
Actually, you can see this in chrome. Go [here](https://reactspectrum.blob.core.windows.net/reactspectrum/cd341f49c118289cea9b5830d6bb13d74b9bda3e/storybook/index.html?path=/story/date-and-time-calendar--min-value&providerSwitcher-express=false), go forward a month, and then tab to the previous button and press Enter. Focus gets lost rather than going back to the calendar, because the onBlur event runs first. Doesn't happen in JSDom or in other browsers.

I fixed this in #4564 by using state instead of a ref.